pytorch_lightning/metrics/functional/classification.py

```diff
@@ -82,7 +82,8 @@ def get_num_classes(
             num_classes = pred.size(1)
         else:
             num_target_classes = int(target.max().detach().item() + 1)
-            num_classes = num_target_classes
+            num_pred_classes = int(pred.max().detach().item() + 1)
```

Comment:
I would raise a warning if user sets `num_classes` and it does not match the max of both
if `num_classes` is not defined then set auto as max of both...
@SkafteNicki @justusschock